test/test_torch.py

```diff
@@ -8334,6 +8334,100 @@ def test_contiguous(self, device):
         x.set_(x.storage(), 0, x.size(), stride)
         self.assertTrue(x.is_contiguous())
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.half, torch.int, torch.long, torch.cfloat, torch.cdouble, torch.bool, torch.uint8)
+    def test_tensor_split_sections(self, device, dtype):
+        input_sizes = [
+            (0,),
+            (10,),
+            (10, 0),
+            (0, 10),
+            (4, 10),
+            (12, 3),
+        ]
+        for input_size in input_sizes:
+            if dtype.is_floating_point or dtype.is_complex:
+                a = torch.randn(*input_size, dtype=dtype, device=device)
+            else:
+                if dtype == torch.bool:
+                    a = torch.randint(2, input_size, dtype=dtype, device=device)
+                else:
+                    a = torch.randint(0, 127, input_size, dtype=dtype, device=device)
+            a_n = a.cpu().numpy()
+            for dim in range(a.dim()):
+                for sections in range(1, 2 * a.size(dim)):
+                    msg = f'input_size {input_size}, sections {sections}, dim {dim}'
+                    result = torch.tensor_split(a, sections, dim)
+                    for result_item in result:
+                        self.assertEqual(result_item.device, torch.device(device), msg=msg)
+                        self.assertEqual(result_item.dtype, dtype, msg=msg)
+                    result_n = np.array_split(a_n, sections, dim)
+                    self.assertEqual(result_n, result, msg=msg)
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.half, torch.int, torch.long, torch.cfloat, torch.cdouble, torch.bool, torch.uint8)
+    def test_tensor_split_indices(self, device, dtype):
+        input_sizes = [
+            (0,),
+            (10,),
+            (10, 0),
+            (0, 10),
+            (4, 10),
+            (12, 3),
+        ]
+        indices_args = [
+            (),
+            (0,),
+            (3,),
+            (10,),
+            (-1,),
+            (-10,),
+            (2, -1),
+            (3, 4, 10),
+            (0, -1, 0, 10),
+        ]
+        for input_size in input_sizes:
+            if dtype.is_floating_point or dtype.is_complex:
+                a = torch.randn(input_size, dtype=dtype, device=device)
+            else:
+                if dtype == torch.bool:
+                    a = torch.randint(2, input_size, dtype=dtype, device=device)
+                else:
+                    a = torch.randint(0, 127, input_size, dtype=dtype, device=device)
+            a_n = a.cpu().numpy()
+            for dim in range(a.dim()):
```

Comment:
dim=-1 here, too

Reply:
I decided to go all the way down to `-a.dim()` just to be sure.